imageproxy.go

```diff
@@ -186,6 +189,21 @@ func (p *Proxy) serveImage(w http.ResponseWriter, r *http.Request) {
 		// pass along the referer header from the original request
 		copyHeader(actualReq.Header, r.Header, "referer")
 	}
+	if p.FollowRedirects {
+		// FollowRedirects is true (default), ensure that the redirected host is allowed
+		p.Client.CheckRedirect = func(newreq *http.Request, via []*http.Request) error {
+			if hostMatches(p.DenyHosts, newreq.URL) || hostMatches(p.AllowHosts, newreq.URL) {
```

Comment:
I don't think this is right... returning an error if the URL is in p.AllowHosts.  See the logic in `p.allowed()` below, which is basically:
 - if URL is in DenyHosts, then error
 - if AllowHosts is not empty and URL is not in AllowHosts, then error

I think we want to do the same thing here...

```go
if hostMatches(p.DenyHosts, newreq.URL) || (len(p.AllowHosts) > 0 && !hostMatches(p.AllowHosts, newreq.URL)) {
  // error
}
```

(or something akin to that)

Reply:
Yes! My fault, I didn't think properly through the logic for AllowHosts 😬

Fixed in d79103e4ccd30edac56ca2d777131e5f4ef55d95